pyega3/libs/data_file.py

```diff
@@ -203,37 +203,34 @@ def download_file_slice(self, file_name, start_pos, length, options=None, pbar=N
 
         self.temporary_files.add(file_name)
 
-        existing_size = os.stat(final_file_name).st_size if os.path.exists(final_file_name) else 0
-        if existing_size > length:
-            os.remove(final_file_name)
-            existing_size = 0
-
-        if pbar:
-            pbar.update(existing_size)
-
-        if existing_size == length:
+        if os.path.exists(final_file_name):
+            existing_size = os.stat(final_file_name).st_size
+            pbar and pbar.update(existing_size)
             return final_file_name
 
+        if os.path.exists(file_name):
+            os.remove(file_name)
+
         try:
-            range_start = start_pos + existing_size
+            range_start = start_pos
             range_end = start_pos + length - 1
             extra_headers = {
                 'Range': f'bytes={range_start}-{range_end}'
             }
 
             with self.data_client.get_stream(path, extra_headers) as r:
                 with open(file_name, 'ba') as file_out:
+                    self.temporary_files.add(file_name)
                     for chunk in r.iter_content(DOWNLOAD_FILE_MEMORY_BUFFER_SIZE):
                         file_out.write(chunk)
-                        if pbar:
-                            pbar.update(len(chunk))
+                        pbar and pbar.update(len(chunk))
 
             total_received = os.path.getsize(file_name)
 
             if total_received != length:
                 raise Exception(f"Slice error: received={total_received}, requested={length}, file='{file_name}'")
 
-        except Exception:
+        except Exception as e:
```

Comment:
But then you don't use the variable `e`. Or was this recommended by Sonar?
(This is just nitpicking, you don't have to create a new commit for this.)

Reply:
yeah, I think forgot to run it for this PR, but last time I check there were no bugs and the quality of this repo is relatively good.